File: src/main/java/wxgj/tinasproutrobot/mirai/test/Echo.java
package wxgj.tinasproutrobot.mirai.test;

import net.mamoe.mirai.Bot;
import net.mamoe.mirai.console.command.CommandSender;
import net.mamoe.mirai.console.command.java.JSimpleCommand;
import net.mamoe.mirai.contact.User;
import wxgj.tinasproutrobot.mirai.JavaPluginMain;

//这样在聊天环境（安装chat-command并分配权限后）发送/echo @<bot> <message>，bot就会复读这个message
public class Echo extends JSimpleCommand {

    public static final Echo INSTANCE = new Echo();

    public Echo() {
        super(JavaPluginMain.INSTANCE, "echo", "复读");
        this.setDescription("复读消息");
    }

    @Handler // 标记这是指令处理器  // 函数名随意
    public void handle(CommandSender sender, User target, String msg) {
        Bot bot = sender.getBot();
        if (bot != null && target.getId() == bot.getId()) { // 判断@对象是否是bot
            sender.sendMessage(msg); // 复读
        }
    }
}

File: src/main/kotlin/command/Echo.kt
package wxgj.tinasproutrobot.mirai.command

import net.mamoe.mirai.console.command.CommandSender
import net.mamoe.mirai.console.command.SimpleCommand
import net.mamoe.mirai.contact.User

object Echo : SimpleCommand(PluginMain.INSTANCE, primaryName = "echo", secondaryNames = arrayOf("复读"), description = "复读消息") {
    @Handler//标记这是指令处理器 //函数名随意
    suspend fun CommandSender.handle(target: User, message: String) {//这两个参数会作为指令参数要求
        if (target.id == bot?.id) {//判断对象是不是@机器人
            sendMessage(message);//复读
        }
    }
}